composer/utils/checkpoint.py

```diff
@@ -532,33 +528,49 @@ def save_checkpoint(
 
     # Sharded checkpoints get their own little folder.
     if state.fsdp_sharded_state_dict_enabled:
+        if using_torch_2():
```

Comment:
suggest adding a comment here explaining why this upleveling is necessary (and link to pytorch docs/issues if any, but fine if that doesn't exist)